scripts/tutorials/gatk/tutorial_8017.wdl

```diff
@@ -0,0 +1,356 @@
+	  OPTICAL_DUPLICATE_PIXEL_DISTANCE=2500 \
+	  ASSUME_SORT_ORDER=queryname
+  }
+ output {
+    File md_bam = "${file_prefix}_md.bam"
+    File md_metrics = "${file_prefix}_md.bam.txt"
+  }
+}
+
+## [4.5] Coordinate sort, fix NM and UQ tags and index for clean BAM
+task SortFixTagsAndIndex {
+  File PICARD
+  File md_bam
+  String file_prefix
+  File ref_fasta
+  File ref_index
+  File ref_dict
+
+  command {
+    set -o pipefail
+    java -jar ${PICARD} SortSam \
+      INPUT=${md_bam} \
+      OUTPUT=/dev/stdout \
+      SORT_ORDER=coordinate | \
+      java -jar ${PICARD} SetNmAndUqTags \
+      INPUT=/dev/stdin \
+      OUTPUT=${file_prefix}_snaut.bam \
+      CREATE_INDEX=true \
+      R=${ref_fasta}
+  }
+ output {
+    File snaut_bam = "${file_prefix}_snaut.bam"
+    File snaut_bai = "${file_prefix}_snaut.bai"
+  }
+}
+
+## [4.6] Call SNP and indel variants in reference confidence (ERC) mode per sample using HaplotypeCaller
+task CallSampleVariants {
+  File GATK
+  String file_prefix
+  File snaut_bam
+  File snaut_bai
+  File ref_fasta
+  File ref_index
+  File ref_dict
+
+  command {
+  java -jar ${GATK} \
+    -T HaplotypeCaller \
+    -R ${ref_fasta} \
+    -I ${snaut_bam} \
+    -o ${file_prefix}.g.vcf \
+	-ERC GVCF \
+	--max_alternate_alleles 3 \
+	--read_filter OverclippedRead \
+	--emitDroppedReads \
+	-bamout ${file_prefix}_hc.bam
+  }
+ output {
+    File sample_gvcfs = "${file_prefix}.g.vcf"
+    File sample_gvcf_indices = "${file_prefix}.g.vcf.idx"
+    File bamout = "${file_prefix}_hc.bam"
+  }
+}
+
+## [4.7] Call genotypes on all samples
+task CallCohortVariants {
+  File GATK
+  Array [File] sample_gvcfs
+  Array [File] sample_gvcf_indices
+  File ref_fasta
+  File ref_index
+  File ref_dict
+  String final_vcf_name
+
+  command {
+  java -jar ${GATK} \
+    -T GenotypeGVCFs \
+    -R ${ref_fasta} \
+    -o ${final_vcf_name}.vcf \
+    --variant ${sep=' --variant ' sample_gvcfs}
+  }
+ output {
+    File cohort_vcf = "${final_vcf_name}.vcf"
+    File cohort_vcf_index = "${final_vcf_name}.vcf.idx"
+  }
+}
+
+# WORKFLOW DEFINITION ########################################################
+
+workflow Tutorial_8017 {
+  Array[Array[File]] input_FASTQ
+  File BWA0715
+  File GATK
+  File PICARD
+  String fastq_directory_path
+  String final_vcf_name
+  String min_unclipped_bases
+  String unmap_contaminant_reads
+  File ref_fasta
+  File ref_index
+  File ref_dict
+  File ref_alt
+  File ref_amb
+  File ref_ann
+  File ref_bwt
+  File ref_pac
+  File ref_sa
+
+## Scatter all but first and last steps
+  scatter (pair in input_FASTQ) {
+
+### Get file_prefix
+	String sub_strip_path = fastq_directory_path
+	String sub_strip_extension = "\\.read.\\.fq" + "$"
+
+### Align FASTQ reads with BWA-MEM
+	call AlignFastqWithBwaMem {
+  	input:
+  	BWA0715 = BWA0715,
+    input_FASTQ1 = pair[0],
+    input_FASTQ2 = pair[1],
+  	file_prefix = sub(sub(pair[0], sub_strip_path, ""), sub_strip_extension, ""),
```

Comment:
I have an array of files and it would be a pain to input file basenames for each of them. This script automatically grabs it from the FASTQ name for you. Less thinking needed and allows for an infinite array of files.